src/lib/data/near.js

```diff
@@ -71,6 +73,98 @@ const apiCall = async (config, methodName, args, blockId, fallback) => {
   }
 };
 
+function getKeyStoreForContract(contractId) {
+  return new nearAPI.keyStores.BrowserLocalStorageKeyStore(
+    window.localStorage,
+    `${contractId}:keystore:`
+  );
+}
+
+async function createWalletConnectionForContract(near, contractId) {
+  const keyStore = getKeyStoreForContract(contractId);
```

Comment:
We should probably check somewhere that the contractId doesn't match `social.near` (for the current network) to avoid some random issues.

Reply:
There is already such a check on `functionCall` ( see https://github.com/NearSocial/VM/pull/156/files#diff-cc75698545c1506b569386ec69e64f50d829fc759a7fb175b256b88630782859R179 )

Reply:
Do we check for this in the `ConfirmTransaction`?

Reply:
No. And I believe if call a function on `social.near`, you might be offered `don't ask again`. This could be solved by also adding this check to `eligibleForDontAskAgain`.

Do we want to **not** offer the `don't ask again` feature for calls to `near.social`?

Reply:
Yeah, let's not give this option for `social.near`.